src/art.rs

```diff
@@ -930,7 +912,7 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
                 let mut commit_version = version;
                 if version == 0 {
                     commit_version = curr_version + 1;
-                } else if check_version && curr_version >= version {
+                } else if check_version && curr_version > version {
```

Comment:
It would be great to have a comment on this method on the discussion we had on how it should be used, in case we revisit this in future and don't remember